### ISSUE TITLE
Reachable code is reported unreachable

### ISSUE THREAD

**Issue by yannl35133** [microsoft/pyright#2778]
**Describe the bug**
Some code is reported unreachable while it is.
(I assume) this is the reason why some variables (here, `s`) are declared unbound while being well defined.

**To Reproduce**
```python
def do_stuff(i):
    if i > 1:
        return

    if i > 0:
        do_stuff(2)
        return

    s = ""
    do_stuff(1)
    # Unreachable from here
    ret = lambda s: None
    return ret(s)  # s is unbound
```
Some parts can be changed with the end declared unreachable, but this presentation most clearly terminates.

I'm using the VSCode extension, v.1.1.188

**Comment by erictraut** [comment]
This bug was fixed many versions ago. The latest version of pyright is 1.1.202, so you're 20 versions out of date. Please confirm that you're running the latest version before filing a bug report.

**Comment by yannl35133** [comment]
I'm using the latest version available in the VSCodium repo

**Comment by erictraut** [comment]
I'm not sure who maintains that. Please let them know that it's way out of date. If they're not going to mirror the latest version of pyright, it would be better for them not to redistribute it.
